synapse/storage/databases/main/roommember.py

```diff
@@ -952,35 +975,23 @@ async def get_current_hosts_in_room(self, room_id: str) -> List[str]:
             sorted by join with the lowest depth first).
         """
 
-        # First we check if we already have `get_users_in_room` in the cache, as
-        # we can just calculate result from that
-        users = self.get_users_in_room.cache.get_immediate(
-            (room_id,), None, update_metrics=False
-        )
-        if users is None and isinstance(self.database_engine, Sqlite3Engine):
+        if isinstance(self.database_engine, Sqlite3Engine):
             # If we're using SQLite then let's just always use
             # `get_users_in_room` rather than funky SQL.
             users = await self.get_users_in_room(room_id)
 
-        if users is not None:
-            # Because `users` is sorted from lowest -> highest depth, the list
-            # of domains will also be sorted that way.
-            domains: List[str] = []
-            # We use a `Set` just for fast lookups
-            domain_set: Set[str] = set()
+            domains: Set[str] = set()
             for u in users:
                 if ":" not in u:
                     continue
                 domain = get_domain_from_id(u)
-                if domain not in domain_set:
-                    domain_set.add(domain)
-                    domains.append(domain)
-            return domains
+                domains.add(domain)
+            return list(domains)
```

Comment:
Might it be more clear if we simply return `get_current_hosts_in_room(room_id)` if we're on SQLite, given that we make no ordering guarantees?